hudi-cli/src/main/java/org/apache/hudi/cli/utils/SparkUtil.java

```diff
@@ -61,9 +62,14 @@ public static SparkLauncher initLauncher(String propertiesFile) throws URISyntax
   }
 
   public static JavaSparkContext initJavaSparkConf(String name) {
+    return initJavaSparkConf(name, Option.empty(), Option.empty());
+  }
+
+  public static JavaSparkContext initJavaSparkConf(String name, Option<String> master,
+                                                   Option<String> executorMemory) {
     SparkConf sparkConf = new SparkConf().setAppName(name);
 
-    String defMasterFromEnv = sparkConf.getenv("SPARK_MASTER");
+    String defMasterFromEnv = master.orElse(sparkConf.getenv("SPARK_MASTER"));
```

Comment:
Can we extract the string literals in this method to be constant fields?

Reply:
> Can we extract the string literals in this method to be constant fields?

@yanghua There is constant fields `DEFAULT_SPARK_MASTER` for master.

Reply:
Actually, I mean, we should avoid using string literals e.g. `SPARK_MASTER `, `spark.hadoop.mapred.output.compress` and so on. WDYT?

Reply:
@yanghua yes, necessary, how about add a file `HoodieCliSparkConfig`.

Reply:
+1 to introduce `HoodieCliSparkConfig`